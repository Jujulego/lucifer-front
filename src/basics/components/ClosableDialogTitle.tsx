import React, { ReactNode } from 'react';

import {
  DialogTitle as MuiDialogTitle,
  IconButton, Typography
} from '@material-ui/core';
import { makeStyles, Theme } from '@material-ui/core/styles';
import CloseIcon from '@material-ui/icons/Close';

import { StyledProps } from 'utils/style';

// Types
export type ClosableDialogTitleClassKey = 'root' | 'title' | 'closeButton'
export type ClosableDialogTitleProps = StyledProps<ClosableDialogTitleClassKey> & {
  children: ReactNode;
  onClose?: () => void;
}

// Styles
const useStyles = makeStyles<Theme, ClosableDialogTitleClassKey>({
  root: {
    display: 'flex',
  },
  title: {
    flexGrow: 1,
  },
  closeButton: {
    marginTop: -8,
    marginBottom: -8,
  },
});

// Component
const ClosableDialogTitle = (props: ClosableDialogTitleProps) => {
  // Props
  const {
    children,
    onClose = () => {}
  } = props;

  // Render
  const styles = useStyles(props);

  return (
    <MuiDialogTitle classes={{ root: styles.root }} disableTypography>
      <Typography
        classes={{ root: styles.title }}
        component="h2" variant="h6"
      >
        { children }
      </Typography>
      <IconButton classes={{ root: styles.closeButton }} edge="end" onClick={onClose}>
        <CloseIcon />
      </IconButton>
    </MuiDialogTitle>
  );
};

export default ClosableDialogTitle;
