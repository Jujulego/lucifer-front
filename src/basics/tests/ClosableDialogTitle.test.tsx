import React from 'react';
import { cleanup, render, act, fireEvent } from '@testing-library/react';

import ClosableDialogTitle from '../components/ClosableDialogTitle';

// Setup
afterEach(cleanup);

// Tests
it('should show a title followed by a button', () => {
  // Render
  const { container } = render(
    <ClosableDialogTitle>Title</ClosableDialogTitle>
  );

  // Check elements
  expect(container).toMatchSnapshot();
});

it('should react to button click', () => {
  const spy = jest.fn();

  // Render
  const { container } = render(
    <ClosableDialogTitle onClose={spy}>Title</ClosableDialogTitle>
  );

  // Test event
  const button = container.querySelector('button')!;
  act(() => { fireEvent.click(button); });

  expect(spy).toHaveBeenCalledTimes(1);
});
