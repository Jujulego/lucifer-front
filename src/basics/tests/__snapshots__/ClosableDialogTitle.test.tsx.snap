// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show a title followed by a button 1`] = `
<div>
  <div
    class="MuiDialogTitle-root makeStyles-root-1"
  >
    <h2
      class="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
    >
      Title
    </h2>
    <button
      class="MuiButtonBase-root MuiIconButton-root makeStyles-closeButton-3 MuiIconButton-edgeEnd"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
